tests/rptest/tests/schema_registry_test.py

```diff
@@ -1223,6 +1224,118 @@ def move_leader():
+        # check the schema that maps to the id
+        def check_each_schema(subject: str, schemas: list[str],
+                              schema_ids: list[int],
+                              subject_versions: list[int]):
+            for version in subject_versions:
+                self.logger.debug(
+                    f"Fetch schema version {version} on subject {subject}")
+                result_raw = self._get_subjects_subject_versions_version(
+                    subject=subject, version=version)
+                self.logger.debug(result_raw)
+                assert result_raw.status_code == requests.codes.ok
+                res = result_raw.json()
+                self.logger.debug(res)
+                assert type(res) == type({})
+                assert "subject" in res
+                assert "version" in res
+                assert "id" in res
+                assert "schema" in res
+                assert res["subject"] == subject
+                assert res["version"] == version
+                assert res["id"] in schema_ids
+                schema = json.loads(schemas[version - 1])
+                assert res["schema"] == schema["schema"]
+
+        admin = Admin(self.redpanda)
+
+        num_subjects = 3
+        subjects = {}
+        for _ in range(num_subjects):
+            subjects[f"{create_topic_names(1)[0]}-key"] = {
+                "schemas": [],
+                "schema_ids": [],
+                "subject_versions": []
+            }
+
+        self.logger.debug("Register and check schemas before restart")
+        for subject in subjects:
+            schemas, schema_ids = register_schemas(subject)
+            result_raw = self._get_subjects_subject_versions(subject=subject)
+            self.logger.debug(result_raw)
+            assert result_raw.status_code == requests.codes.ok
+            subject_versions = result_raw.json()
+            check_subject_versions(subject, subject_versions)
+            check_each_schema(subject, schemas, schema_ids, subject_versions)
+            subjects[subject]["schemas"] = schemas
+            subjects[subject]["schema_ids"] = schema_ids
+            subjects[subject]["subject_versions"] = subject_versions
+        check_subjects(list(subjects.keys()))
+
+        self.logger.debug("Restart the schema registry")
+        result_raw = admin.redpanda_services_restart(
+            rp_service='schema-registry')
+        self.logger.debug(result_raw)
+        assert result_raw.status_code == requests.codes.ok
+
+        self.logger.debug("Check schemas after restart")
+        check_subjects(list(subjects.keys()))
+        for subject in subjects:
+            check_subject_versions(subject, subjects[subject]["schema_ids"])
+            check_each_schema(subject, subjects[subject]["schemas"],
+                              subjects[subject]["schema_ids"],
+                              subjects[subject]["subject_versions"])
```

Comment:
what is it here that we are testing for?

after a restart do we expect previous state to be restored? what state do we expect? if we expect it to be restored, is there a race condition?

how do we know that the restart occurred based on this test?

Reply:
> what is it here that we are testing for?

Subjects, versions, ids, and schema are the same as before: https://github.com/redpanda-data/redpanda/pull/7765#pullrequestreview-1236217441

> after a restart do we expect previous state to be restored?

Yes.

> what state do we expect?

Same as before (identifiers should be the same)

> if we expect it to be restored, is there a race condition?

No, requests aren't served until the last offset is read from the topic.

We could opportunistically load the `_schemas` topic after restart, instead of waiting for the first request.

> how do we know that the restart occurred based on this test?

Good question.

Reply:
Added the logging in https://github.com/redpanda-data/redpanda/pull/8127/commits/b7705b6e90c31d724af2015728eed0d3ee8060fb

Added log checking into the DT tests in https://github.com/redpanda-data/redpanda/pull/8127/commits/747d13a8678e7ea628a3972add8686d2b2568b77